message_app/user_sign_in.py

```diff
@@ -0,0 +1,43 @@
+from flask import Blueprint, request, session, redirect, url_for
+from message_app.model import User
+from hashlib import sha256
+
+from message_app.utils import hashing
+
+user_sign_in = Blueprint("user_sign_in", __name__)
+
+@user_sign_in.route("/api/signin", methods=["POST"])
+def api_user_signin():
+    try:
+        params = request.form
+        username = params.get("username", "")
+        input_password = params.get("password", "")
+
+        if username == "" or input_password == "":
+            session["error"] = "Invalid login credentials. Please try again."
+            raise Exception(session["error"])
+
+        # Connect and fetch data from the users table
+        user_data = User.select(username)
+
+        # User not found
+        if not user_data:
+            session["error"] = "Invalid login credentials. Please try again."
+            raise Exception(session["error"])
+
+        user_password = user_data.password_hash
+        salt = user_data.password_salt
+
+        # Hash the password
+        password = hashing(input_password, salt).password_hash
+
+        if password == user_password:
```

Comment:
Based on my suggestion above, this could be `if user_password_input_hash == user_password_hash`